### PR TITLE
Mobile (Plan 19, spike A): gate desktop-only capabilities; crate compiles for iOS

### PR DESCRIPTION
**Pull request by maccman**
## Summary

[Plan 19](docs/plans/19-mobile.md) gate **spike A — both the compile and simulator halves**: the desktop crate cross-compiles for iOS, and the app boots on the iPhone 17 Pro simulator with all four native-capability probes passing.

- **Cargo.toml**: `fastembed`/`hf-hub`, `notify`/`notify-debouncer-full`, `trash`, and `tauri-plugin-window-state` move under the existing `cfg(not(ios/android))` target section (joining updater/process).
- **Mobile stand-ins** ([watcher_mobile.rs](apps/desktop/src-tauri/src/watcher_mobile.rs), [embed_mobile.rs](apps/desktop/src-tauri/src/embed_mobile.rs)): identical command surface selected by `#[cfg]` + `#[path]`, so `lib.rs`'s single `invoke_handler` list needs no platform branches. The embed commands fail loudly (`AppError::Unknown`); the watcher commands are documented no-ops — the shared graph-index lifecycle awaits `watch_start` before going live, so erroring would cut the in-process event path off from the indexer (caught by Bugbot).
- **Deletes on mobile** move to graph-local `.reflect/trash/` (timestamp-suffixed on collision).
- **Capabilities**: `window-state:default` moves to the platform-scoped `desktop.json`.
- **iOS link fix**: libgit2 (vendored into `libapp.a`) needs `libz.tbd` + `libiconv.tbd` declared in [ios.project.yml](apps/desktop/src-tauri/ios.project.yml) — cargo's `rustc-link-lib` directives don't reach the final Xcode link. `gen/apple/` regenerated via `tauri ios init`.
- **Temporary spike instrumentation** ([spike_mobile.rs](apps/desktop/src-tauri/src/spike_mobile.rs), `cfg(mobile)`): startup probes for the keychain, FTS5, `Documents/` IO, and libgit2 — kept until the physical-device half of the gate runs.

## Verification

- `cargo check -p reflect-open --target aarch64-apple-ios` passes clean.
- `tauri ios dev` on the iPhone 17 Pro simulator: **BUILD SUCCEEDED**, app launches, frontend renders (GraphChooser over the dev server), and all four probes log `[plan19-spike] PASS` — keychain round-trip, SQLite FTS5 MATCH, `Documents/` read/write, libgit2 init+commit.
- Desktop unchanged: all 99 crate tests pass, `pnpm check` passes.

- **Plan 19 steps 3–4 (same PR, continued)**: `app_platform` + `mobile_graph_root` commands with zod-validated core wrappers; Files-app exposure + `app.reflect.ios` identity; a lazy `PlatformRoot` gate splitting desktop chrome ([desktop-root.tsx](apps/desktop/src/desktop-root.tsx)) from the mobile tree; fixed-root graph bootstrap in the shared `GraphProvider` (derived fresh every launch — container paths are not stable); and a mobile Today screen mounting the real editor via `NotePane`. **Verified end-to-end on the simulator: typing into meowdown lands the daily note on disk through the shared save pipeline.** The document stack requires `RouterProvider` (white-screen unmount without it — now guarded by `MobileErrorBoundary`).

- **Plan 19 step 5 + decision 8 (same PR, continued)**: the mobile write-event seam (`setLocalWriteEcho` — local writes emit their in-process change batch so reindex/invalidation/sync dirty-marking work without a watcher; unit-tested) and the first-party **keyboard plugin** ([plugins/tauri-plugin-keyboard](plugins/tauri-plugin-keyboard)): Swift streams keyboard overlap as `keyboardChange` events and disables the system scroll nudge (tauri#9907); the frontend mirrors it into `--keyboard-height` and Today yields via `max(safe-area, keyboard)`. **Simulator-verified with the software keyboard: the caret line stays visible above the keyboard while editing, and the note saves to disk.**

## Next (not in this PR)

The physical-device half of spike A (same probes on an iPhone), then gate spike B (keyboard plugin + meowdown editing on-device). The mobile UI itself starts after both gates.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Broad platform `cfg` splits and new mobile graph/delete/IPC paths could regress desktop or miss edge cases; iOS/Xcode and plugin ACL wiring add build-time failure modes despite desktop tests staying green.
> 
> **Overview**
> This PR lands **Plan 19** mobile groundwork: the Tauri crate cross-compiles for iOS, the app boots on simulator with native probes, and a first mobile UI path edits today’s daily note end-to-end.
> 
> **Rust / iOS build:** Desktop-only deps (`fastembed`, `notify`, `trash`, `window-state`, updater) move behind `cfg(not(ios/android))`; **mobile stand-ins** keep the same IPC for the file watcher (no-op) and embeddings (fail loud). Deletes on mobile go to **`.reflect/trash/`** instead of OS trash. New commands **`app_platform`** and **`mobile_graph_root`**; temporary **`spike_mobile`** startup probes (keychain, FTS5, Documents IO, libgit2). iOS project identity becomes **`app.reflect.ios` / Reflect**, Files-app sharing, **`libz`/`libiconv`** link fix, and **`capabilities/mobile.json`** for the keyboard plugin.
> 
> **Frontend:** Lazy **`PlatformRoot`** splits **`DesktopRoot`** vs **`MobileRoot`**. Mobile auto-opens a **fresh `Documents/` root** (no recents), **`setLocalWriteEcho`** so writes drive index/sync without a watcher, and a **Today** screen with real **`NotePane`**. **`tauri-plugin-keyboard`** streams overlap height; layout uses **`--keyboard-height`** for avoidance.
> 
> Desktop behavior is intended unchanged; risk is mostly new platform branches and a large touch surface in one PR.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 132bbb3a3aa95584f1eff13993b6e419e60b6733. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/desktop/src-tauri/src/watcher_mobile.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#[derive(Default)]
+pub struct WatcherState;
+
+/// Mobile has no watcher: callers must not rely on `index:changed` events
+/// from the shell. Fails loudly — the mobile frontend never calls this.
+#[tauri::command]
+pub fn watch_start(_watcher: State<WatcherState>) -> AppResult<()> {
+    Err(AppError::Unknown {
+        message: "the file watcher is desktop-only".into(),
+    })
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src-tauri/src/watcher_mobile.rs]: Fixed in bf89a30 — `watch_start` is now a documented no-op (`Ok(())`) on mobile: the contract is "start platform file watching, if any", and erroring would tear down the pending index subscription in `graph-index.ts` exactly as you described. Good catch.

File: apps/desktop/src-tauri/gen/apple/reflect-open.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -215,6 +325,7 @@
 				F00F5EAED90E34102F9AE651 /* Assets.xcassets in Resources */,
 				5693613DE956199D6486EA26 /* LaunchScreen.storyboard in Resources */,
 				C1FBC0B57C41A13D88225DFB /* assets in Resources */,
+				FF9BC489B3A9BE06C2125BAD /* libapp.a in Resources */,
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src-tauri/gen/apple/reflect-open.xcodeproj/project.pbxproj]: Fixed in ee48c99 — the cause was xcodegen sweeping Externals/ build artifacts into the resources phase when regenerating after a build; the template now pins `buildPhase: none` on Externals so regeneration is deterministic. Verified the simulator build succeeds with no CpResource step for libapp.a.

File: apps/desktop/src-tauri/src/fs/mod.rs [hunk 3]
```diff
@@ -294,11 +295,46 @@ pub(crate) fn move_note_file(root: &Path, from: &str, to: &str) -> AppResult<()>
...
+        let stem = name
+            .file_stem()
+            .and_then(|value| value.to_str())
+            .unwrap_or("note");
+        target = trash_dir.join(match name.extension().and_then(|value| value.to_str()) {
+            Some(ext) => format!("{stem}-{millis}.{ext}"),
+            None => format!("{stem}-{millis}"),
+        });
+    }
+    fs::rename(abs, target)?;
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src-tauri/src/fs/mod.rs:347]: Fixed in f94a6fe — the suffix now loops (millis, then millis-2, millis-3, …) until the trash name is free, so repeat deletes of the same filename can't fail the rename.
